wormhole-connect/src/utils/AsyncCache.ts

```diff
@@ -0,0 +1,58 @@
+interface CacheEntry<T> {
+  value: T;
+  timestamp: number;
+  isError?: boolean;
+}
+
+export class AsyncCache<T> {
+  private cache = new Map<string, CacheEntry<T>>();
+  private pendingRequests = new Map<string, Promise<T>>();
+
+  constructor(private ttl: number) {}
```

Comment:
Minor nit: can we append the units here? It's not obvious to me if this should be seconds or milliseconds from reading the signature

Reply:
Sure thing